### PR TITLE
test: add test with multiple ORM schemas

### LINKED ISSUE #618: [Tests] use 2 schemas in the same database
after merging https://github.com/zenstruck/foundry/pull/615, in order to prevent any regressions in "reset database" behavior, I think we should reproduce in test a case where 2 schemas exist in the same database

ping @vincentchalamon

### PR DESCRIPTION
**Pull request by vincentchalamon**
Fixes #618

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -29,6 +30,11 @@
     $fs->remove(__DIR__.'/Fixture/Migrations');
     $fs->mkdir(__DIR__.'/Fixture/Migrations');
 
+    // restore custom migrations
```
Comment (kbond) [inline tests/bootstrap.php]: Can we use `Symfony\Component\Filesystem\Filesystem` to just copy the entire directory?

**Review by kbond — COMMENTED** [review]
This looks great, thanks for working on this Vincent!

Regarding the failing CI (I'm not familiar with multiple schemas). Do you know what's going on there?

File: tests/bootstrap.php (hunk 1 above)
Comment (vincentchalamon) [reply tests/bootstrap.php]: Filesystem only allows to copy files, not directories

**Comment by vincentchalamon** [comment]
Hi @kbond,

> Regarding the failing CI (I'm not familiar with multiple schemas). Do you know what's going on there?

The `doctrine:schema:update` command doesn't seem to support multiple schemas. To handle such specific use case, I had to separate `migrate` mode from `schema` mode fixtures. Push in progress.

This PR is still in WIP till I find a clean solution to test this specific use case.

File: tests/bootstrap.php (hunk 1 above)
Comment (kbond) [reply tests/bootstrap.php]: `Filesystem::mirror()` should do the trick (but this isn't a big deal)

**Review by kbond — APPROVED** [review]

**Comment by nikophil** [comment]
Hi @vincentchalamon 

I'm wondering if a simpler solution would not be to declare two entity managers, affect one entity to this entity manager, and configure the test kernel to not reset this specific EM. WDYT?

**Comment by vincentchalamon** [comment]
> Hi @vincentchalamon
> 
> I'm wondering if a simpler solution would not be to declare two entity managers, affect one entity to this entity manager, and configure the test kernel to not reset this specific EM. WDYT?

I'm afraid using multiple entity managers won't solve the original issue. The idea is to add non-regression tests to ensure the database is fully and properly removed before running the migrations (cf. https://github.com/zenstruck/foundry/pull/615).

The idea here is to have a custom migration which should fail the tests if not run properly (for instance, if `AbstractORMPersistenceStrategy::resetSchema` method is updated).

**Comment by nikophil** [comment]
ok thanks for explanations, I understand better now!

**Review by nikophil — APPROVED** [review]

**Pull request merged**